disnake/threads.py

```diff
@@ -952,3 +1072,154 @@ def _from_data(self, data: ThreadMemberPayload):
     def thread(self) -> Thread:
         """:class:`Thread`: The thread this member belongs to."""
         return self.parent
+
+
+class ForumTag(Hashable):
+    """
+    Represents a tag for threads in forum channels.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two tags are equal.
+
+        .. describe:: x != y
+
+            Checks if two tags are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the tag's hash.
+
+        .. describe:: str(x)
+
+            Returns the tag's name.
+
+    .. versionadded:: 2.6
+
+
+    Examples
+    --------
+
+    Creating a new tag:
+
+    .. code-block:: python3
+
+        tags = forum.available_tags
+        tags.append(ForumTag(name="cool new tag", moderated=True))
+        await forum.edit(available_tags=tags)
+
+    Editing an existing tag:
+
+    .. code-block:: python3
+
+        tags = []
+        for tag in forum.available_tags:
+            if tag.id == 1234:
+                tag = tag.with_changes(name="whoa new name")
+            tags.append(tag)
+        await forum.edit(available_tags=tags)
+
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The tag's ID. Note that if this tag was manually constructed,
+        this will be ``0``.
+    name: :class:`str`
+        The tag's name.
+    moderated: :class:`bool`
+        Whether only moderators can add this tag to threads or remove it.
+        Defaults to ``False``.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`]]
+        The emoji associated with this tag, if any.
+        Note that due to a Discord limitation, this will have an empty
+        :attr:`~PartialEmoji.name` if it is a custom :class:`PartialEmoji`.
+    """
+
+    __slots__ = ("id", "name", "moderated", "emoji")
+
+    def __init__(
+        self,
+        name: str,
+        emoji: Optional[Union[str, PartialEmoji, Emoji]] = None,
+        *,
```

Comment:
```suggestion
        *,
        name: str,
        emoji: Optional[Union[str, PartialEmoji, Emoji]] = None,
```

Reply:
I agree with making `emoji` kwarg-only, but `name` is fine, imo - `ForumTag("solved")` seems fairly intuitive to me.